snap/errors_test.go

```diff
@@ -47,3 +47,25 @@ func (s *errorsSuite) TestNotInstalledErrorIs(c *C) {
 	c.Check(err.Is(&snap.NotInstalledError{}), Equals, true)
 	c.Check(err.Is(errors.New("some error")), Equals, false)
 }
+
+func (s *errorsSuite) TestComponentNotInstalledErrorDetails(c *C) {
+	err := snap.ComponentNotInstalledError{
+		NotInstalledError: snap.NotInstalledError{Snap: "mysnap", Rev: snap.R(1)},
+		Component:         "mycomp",
+		CompRev:           snap.R(7),
+	}
+	c.Check(err, ErrorMatches,
+		`revision 7 of component "mycomp" is not installed for revision 1 of snap "mysnap"`)
+
+	err = snap.ComponentNotInstalledError{
+		NotInstalledError: snap.NotInstalledError{Snap: "mysnap", Rev: snap.R(1)},
+		Component:         "mycomp",
+	}
+	c.Check(err, ErrorMatches, `component "mycomp" is not installed for revision 1 of snap "mysnap"`)
+}
+
+func (s *errorsSuite) TestComponentNotInstalledErrorDetailsIs(c *C) {
+	err := snap.ComponentNotInstalledError{}
+	c.Check(err.Is(&snap.ComponentNotInstalledError{}), Equals, true)
+	c.Check(err.Is(errors.New("some error")), Equals, false)
```

Comment:
Testing this via `errors.Is` rather than directly calling the `Is` method is probably more representative of how it will actually be used.

Reply:
This was for testing `ComponentNotInstalledError.Is(error)`, which I created just because there was also a `NotInstalledError.Is(error)`, but in the end the latter is not used anymore, so I have removed the new and the old one in the end.

Reply:
This would use `NotInstalledError.Is`, right? https://github.com/snapcore/snapd/blob/master/overlord/devicestate/devicestate.go?plain=1#L530